### PR TITLE
Removed idist.barrier() where needed

### PR DESCRIPTION
**Pull request by sayantan1410**
Related to #168 

### Description
Remove idist.barrier() if no distribution configuration is specified. 

### Additional context
I have tried to solve the problem with what I understood, incase my understanding is completely wrong, kindly correct me and I will update the PR. 

### What is the purpose of this pull request?

- [x] Bug fix

### CODE REVIEW

**Comment by vfdev-5** [comment]
@sayantan1410 thanks for the PR, but the idea is to make the code you removed configurable and not just removing it. 
In the app you have UI option to choose or not distributed computation, the idea is to add this code if the distributed option is activated.

**Comment by sayantan1410** [comment]
Thank you for the clarification, just another question how to check if distributed configuration is selected in the UI ?

**Comment by ydcjeff** [comment]
@sayantan1410 You can use `it.use_dist` in python code comment to check if distributed option is selected or not. For writing JS template code in python comment, check https://raw.githubusercontent.com/pytorch-ignite/code-generator/main/src/templates/template-vision-dcgan/README.md

JS template code is in [ejs](https://ejs.co/) format and we use `#` for open and close delimiters and `:::` for normal delimiter. So, you need to write in python code like:

```py
# below line is JS template code
#::: if (it.use_dist) { :::#
# setting local rank code
#::: } :::#
```

**Comment by sayantan1410** [comment]
Thank you so much will check it out !

**Comment by sayantan1410** [comment]
I have made the changes as per my understanding, let me if it needs some correction, also do I need to run tests with pytest, or there is some other method, I have made the set up successfully and the app is running locally.

**Comment by ydcjeff** [comment]
@sayantan1410 Changes look good, can you rebase?

**Comment by sayantan1410** [comment]
Hey, I have rebased the main branch to this branch, hopefully that is what you were looking for, if I have made some mistakes let me know and I will correct it.

File: netlify.toml
Comment (vfdev-5) [inline netlify.toml:1]: Why this is removed , @sayantan1410 ?

Comment (vfdev-5) [reply netlify.toml:1]: Ok, I see, @ydcjeff can you put this change to main and not in the PR ?

**Comment by sayantan1410** [comment]
Okay putting it in the main branch and making a PR from that branch okay ?

File: netlify.toml
Comment (ydcjeff) [reply netlify.toml:1]: It is already removed in main. https://github.com/pytorch-ignite/code-generator/commit/5954335fc5ef71e42da9ba81b24e74388c621e11

**Comment by ydcjeff** [comment]
@sayantan1410 In case if you have problem in rebase, you can turn on `Allow edits and access to secrets by maintainers` on the right end side so that we can push to your branch.

Also can you run `bash scripts/run_code_style.sh lint`, it is failing at https://github.com/pytorch-ignite/code-generator/runs/4625302143?check_suite_focus=true

**Comment by sayantan1410** [comment]
Okay thank you !!

**Comment by sayantan1410** [comment]
I tried running

> Also can you run `bash scripts/run_code_style.sh lint`

and the output was " 4 files would be reformatted 25 left unchanged", but there was no change in any code. 
I am making another pull request from the main branch that probably would solve the  problem. 
Also, 
Merry Christmas :)

**Comment by ydcjeff** [comment]
@sayantan1410 Sorry. It should be `bash scripts/run_code_style.sh fmt`. That will format the code automatically.

**Comment by sayantan1410** [comment]
Okay trying it !! Thank you. @ydcjeff

**Comment by sayantan1410** [comment]
Hey, I made another commit after running `bash scripts/run_code_style.sh fmt`  , Let me know if this works !! Thank you.

**Comment by vfdev-5** [comment]
@sayantan1410 I'm updating black version in #195, once it is merged please make sure to install this version and reformat the code again:
```
bash scripts/run_code_style.sh install
bash scripts/run_code_style.sh fmt
```

File: src/templates/template-vision-classification/data.py
Comment (vfdev-5) [inline src/templates/template-vision-classification/data.py:15]: Let's also put `it.use_dist` to local_rank definition. In non-dist config there is no need to define it

**Comment by vfdev-5** [comment]
@sayantan1410 thanks for the PR and updates! In this PR you are updating only one template, but ultimately we would like to fix all the template in this way. We can do this either in this PR or in follow-up PRs (in this case please modify `fix #168` by `related to #168` as this PR does not close completely the issue).

**Comment by sayantan1410** [comment]
I will make another PR and will do this for all the templates. Thank you for merging, but still one case was not successful can you explain why that happened and how to get rid of it ??

**Comment by vfdev-5** [comment]
@sayantan1410 remains to do :
- https://github.com/pytorch-ignite/code-generator/pull/194#discussion_r775173093
- https://github.com/pytorch-ignite/code-generator/pull/194#issuecomment-1001075457

**Comment by sayantan1410** [comment]
> @sayantan1410 remains to do :
> 
> * [Removed idist.barrier() where needed #194 (comment)](https://github.com/pytorch-ignite/code-generator/pull/194#discussion_r775173093)
> * [Removed idist.barrier() where needed #194 (comment)](https://github.com/pytorch-ignite/code-generator/pull/194#issuecomment-1001075457)

Sorry for the delay, I am making the change today and pushing it in this PR only. Okay ??

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks @sayantan1410

**Comment by sayantan1410** [comment]
Thank you so much for merging the PR. I will make the necessary changes for the other templates as well ASAP. Should I make the PR form the main branch only or from a new branch. What is your suggestion ? Thank you @vfdev-5

**Comment by vfdev-5** [comment]
@sayantan1410 sounds good for other templates, thanks!
New PRs should be done to current main branch from your fork branch synced with our main.

**Comment by sayantan1410** [comment]
Thank you so much I will follow this !! @vfdev-5

**Pull request merged**
